


<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN"
  "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<html id="htmlId">
<head>
  <title>Coverage Report :: ConcurrentModificationException</title>
  <style type="text/css">
    @import "../../.css/coverage.css";
  </style>
</head>

<body>
<div class="header"></div>

<div class="content">
<div class="breadCrumbs">
    [ <a href="../../index.html">all classes</a> ]
    [ <a href="../index.html">java.util</a> ]
</div>

<h1>Coverage Summary for Class: ConcurrentModificationException (java.util)</h1>

<table class="coverageStats">
<tr>
  <th class="name">Class</th>
<th class="coverageStat 
">
  Class, %
</th>
<th class="coverageStat 
">
  Method, %
</th>
<th class="coverageStat 
">
  Line, %
</th>
</tr>
<tr>
  <td class="name">ConcurrentModificationException</td>
<td class="coverageStat">
  <span class="percent">
    0%
  </span>
  <span class="absValue">
    (0/ 1)
  </span>
</td>
<td class="coverageStat">
  <span class="percent">
    0%
  </span>
  <span class="absValue">
    (0/ 4)
  </span>
</td>
<td class="coverageStat">
  <span class="percent">
    0%
  </span>
  <span class="absValue">
    (0/ 4)
  </span>
</td>
</tr>

</table>

<br/>
<br/>


<div class="sourceCode"><i>1</i>&nbsp;/*
<i>2</i>&nbsp; * Copyright (c) 1997, 2013, Oracle and/or its affiliates. All rights reserved.
<i>3</i>&nbsp; * ORACLE PROPRIETARY/CONFIDENTIAL. Use is subject to license terms.
<i>4</i>&nbsp; *
<i>5</i>&nbsp; *
<i>6</i>&nbsp; *
<i>7</i>&nbsp; *
<i>8</i>&nbsp; *
<i>9</i>&nbsp; *
<i>10</i>&nbsp; *
<i>11</i>&nbsp; *
<i>12</i>&nbsp; *
<i>13</i>&nbsp; *
<i>14</i>&nbsp; *
<i>15</i>&nbsp; *
<i>16</i>&nbsp; *
<i>17</i>&nbsp; *
<i>18</i>&nbsp; *
<i>19</i>&nbsp; *
<i>20</i>&nbsp; *
<i>21</i>&nbsp; *
<i>22</i>&nbsp; *
<i>23</i>&nbsp; *
<i>24</i>&nbsp; */
<i>25</i>&nbsp;
<i>26</i>&nbsp;package java.util;
<i>27</i>&nbsp;
<i>28</i>&nbsp;/**
<i>29</i>&nbsp; * This exception may be thrown by methods that have detected concurrent
<i>30</i>&nbsp; * modification of an object when such modification is not permissible.
<i>31</i>&nbsp; * &lt;p&gt;
<i>32</i>&nbsp; * For example, it is not generally permissible for one thread to modify a Collection
<i>33</i>&nbsp; * while another thread is iterating over it.  In general, the results of the
<i>34</i>&nbsp; * iteration are undefined under these circumstances.  Some Iterator
<i>35</i>&nbsp; * implementations (including those of all the general purpose collection implementations
<i>36</i>&nbsp; * provided by the JRE) may choose to throw this exception if this behavior is
<i>37</i>&nbsp; * detected.  Iterators that do this are known as &lt;i&gt;fail-fast&lt;/i&gt; iterators,
<i>38</i>&nbsp; * as they fail quickly and cleanly, rather that risking arbitrary,
<i>39</i>&nbsp; * non-deterministic behavior at an undetermined time in the future.
<i>40</i>&nbsp; * &lt;p&gt;
<i>41</i>&nbsp; * Note that this exception does not always indicate that an object has
<i>42</i>&nbsp; * been concurrently modified by a &lt;i&gt;different&lt;/i&gt; thread.  If a single
<i>43</i>&nbsp; * thread issues a sequence of method invocations that violates the
<i>44</i>&nbsp; * contract of an object, the object may throw this exception.  For
<i>45</i>&nbsp; * example, if a thread modifies a collection directly while it is
<i>46</i>&nbsp; * iterating over the collection with a fail-fast iterator, the iterator
<i>47</i>&nbsp; * will throw this exception.
<i>48</i>&nbsp; *
<i>49</i>&nbsp; * &lt;p&gt;Note that fail-fast behavior cannot be guaranteed as it is, generally
<i>50</i>&nbsp; * speaking, impossible to make any hard guarantees in the presence of
<i>51</i>&nbsp; * unsynchronized concurrent modification.  Fail-fast operations
<i>52</i>&nbsp; * throw {@code ConcurrentModificationException} on a best-effort basis.
<i>53</i>&nbsp; * Therefore, it would be wrong to write a program that depended on this
<i>54</i>&nbsp; * exception for its correctness: &lt;i&gt;{@code ConcurrentModificationException}
<i>55</i>&nbsp; * should be used only to detect bugs.&lt;/i&gt;
<i>56</i>&nbsp; *
<i>57</i>&nbsp; * @author  Josh Bloch
<i>58</i>&nbsp; * @see     Collection
<i>59</i>&nbsp; * @see     Iterator
<i>60</i>&nbsp; * @see     Spliterator
<i>61</i>&nbsp; * @see     ListIterator
<i>62</i>&nbsp; * @see     Vector
<i>63</i>&nbsp; * @see     LinkedList
<i>64</i>&nbsp; * @see     HashSet
<i>65</i>&nbsp; * @see     Hashtable
<i>66</i>&nbsp; * @see     TreeMap
<i>67</i>&nbsp; * @see     AbstractList
<i>68</i>&nbsp; * @since   1.2
<i>69</i>&nbsp; */
<i>70</i>&nbsp;public class ConcurrentModificationException extends RuntimeException {
<i>71</i>&nbsp;    private static final long serialVersionUID = -3666751008965953603L;
<i>72</i>&nbsp;
<i>73</i>&nbsp;    /**
<i>74</i>&nbsp;     * Constructs a ConcurrentModificationException with no
<i>75</i>&nbsp;     * detail message.
<i>76</i>&nbsp;     */
<b class="nc"><i>77</i>&nbsp;    public ConcurrentModificationException() {</b>
<i>78</i>&nbsp;    }
<i>79</i>&nbsp;
<i>80</i>&nbsp;    /**
<i>81</i>&nbsp;     * Constructs a {@code ConcurrentModificationException} with the
<i>82</i>&nbsp;     * specified detail message.
<i>83</i>&nbsp;     *
<i>84</i>&nbsp;     * @param message the detail message pertaining to this exception.
<i>85</i>&nbsp;     */
<i>86</i>&nbsp;    public ConcurrentModificationException(String message) {
<b class="nc"><i>87</i>&nbsp;        super(message);</b>
<i>88</i>&nbsp;    }
<i>89</i>&nbsp;
<i>90</i>&nbsp;    /**
<i>91</i>&nbsp;     * Constructs a new exception with the specified cause and a detail
<i>92</i>&nbsp;     * message of {@code (cause==null ? null : cause.toString())} (which
<i>93</i>&nbsp;     * typically contains the class and detail message of {@code cause}.
<i>94</i>&nbsp;     *
<i>95</i>&nbsp;     * @param  cause the cause (which is saved for later retrieval by the
<i>96</i>&nbsp;     *         {@link Throwable#getCause()} method).  (A {@code null} value is
<i>97</i>&nbsp;     *         permitted, and indicates that the cause is nonexistent or
<i>98</i>&nbsp;     *         unknown.)
<i>99</i>&nbsp;     * @since  1.7
<i>100</i>&nbsp;     */
<i>101</i>&nbsp;    public ConcurrentModificationException(Throwable cause) {
<b class="nc"><i>102</i>&nbsp;        super(cause);</b>
<i>103</i>&nbsp;    }
<i>104</i>&nbsp;
<i>105</i>&nbsp;    /**
<i>106</i>&nbsp;     * Constructs a new exception with the specified detail message and
<i>107</i>&nbsp;     * cause.
<i>108</i>&nbsp;     *
<i>109</i>&nbsp;     * &lt;p&gt;Note that the detail message associated with &lt;code&gt;cause&lt;/code&gt; is
<i>110</i>&nbsp;     * &lt;i&gt;not&lt;/i&gt; automatically incorporated in this exception&#39;s detail
<i>111</i>&nbsp;     * message.
<i>112</i>&nbsp;     *
<i>113</i>&nbsp;     * @param  message the detail message (which is saved for later retrieval
<i>114</i>&nbsp;     *         by the {@link Throwable#getMessage()} method).
<i>115</i>&nbsp;     * @param  cause the cause (which is saved for later retrieval by the
<i>116</i>&nbsp;     *         {@link Throwable#getCause()} method).  (A {@code null} value
<i>117</i>&nbsp;     *         is permitted, and indicates that the cause is nonexistent or
<i>118</i>&nbsp;     *         unknown.)
<i>119</i>&nbsp;     * @since 1.7
<i>120</i>&nbsp;     */
<i>121</i>&nbsp;    public ConcurrentModificationException(String message, Throwable cause) {
<b class="nc"><i>122</i>&nbsp;        super(message, cause);</b>
<i>123</i>&nbsp;    }
<i>124</i>&nbsp;}
</div>
</div>

<div class="footer">
    
    <div style="float:right;">generated on 2018-11-19 23:15</div>
</div>
</body>
</html>
